command/pki_add_intermediate.go

```diff
@@ -0,0 +1,139 @@
+package command
+
+import (
+	"fmt"
+	"github.com/hashicorp/vault/command/pkicli"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+	"strings"
+)
+
+var (
+	_ cli.Command             = (*PKIAddIntermediateCommand)(nil)
+	_ cli.CommandAutocomplete = (*PKIAddIntermediateCommand)(nil)
+)
+
+type PKIAddIntermediateCommand struct {
+	*BaseCommand
+
+	flagMount string
+	flagRootMount string
+	flagCommonName string
+}
+
+func (c *PKIAddIntermediateCommand) Synopsis() string {
+	return "Generate intermediate certificate"
+}
+
+func (c *PKIAddIntermediateCommand) Help() string {
+	helpText := `
+Usage: vault pki add-intermediate [options] [ROOT_MOUNT] PATH COMMON_NAME [K=V]
+
+  Configures an intermediate mount and generate the intermediate certificate.
+  The intermediate certificate is the one from which all leaf certificates will be generated.
+  This intermediate will be signed by the root if root-mount specified in the input parameters.
+  Other parameters can be specified in K=V format (mirroring vault write).
+
+  Configure an intermediate mount at path pki-int with a specific ttl:
+      $ vault pki add-intermediate pki pki-int example.com ttl=48000h
+  
+
+` + c.Flags().Help()
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *PKIAddIntermediateCommand) Flags() *FlagSets {
+	set := c.flagSet(FlagSetHTTP)
+	f := set.NewFlagSet("Command Options")
+
+	f.StringVar(&StringVar{
+		Name:    "mount",
+		Target:  &c.flagMount,
```

Comment:
I don't think this one is necessary? That's what the first positional argument should be, correct? `args[1]` in particular.

Reply:
Yup, updated the code now. Thanks.